### PR TITLE
Fix phpstan and bump Psalm

### PR DESCRIPTION
**Pull request by VincentLanglet**
```markdown
### Changed
- Use `FilterData` instead of `array` in filters.
```

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
         "doctrine/doctrine-bundle": "^2.3",
         "doctrine/orm": "^2.5",
         "doctrine/persistence": "^2.0",
-        "sonata-project/admin-bundle": "^4.0@alpha",
+        "sonata-project/admin-bundle": "^4.0@dev",
```
Comment (jordisala1991) [inline composer.json:32]: you will also need to conflict with previous alphas

Comment (VincentLanglet) [reply composer.json:32]: "sonata-project/admin-bundle": "<=4.0.0-alpha-1" ?

**Comment by VincentLanglet** [comment]
Works with https://github.com/sonata-project/SonataAdminBundle/pull/7092

**Comment by VincentLanglet** [comment]
I'll wait for https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1425 since I use `Paginator` as return type in the last commit.

**Comment by VincentLanglet** [comment]
This is ready to be merged @sonata-project/contributors

**Review by jordisala1991 — APPROVED** [review]

**Comment by jordisala1991** [comment]
Probably a changelog would be needed.

**Comment by VincentLanglet** [comment]
> Probably a changelog would be needed.

For which change ?

**Comment by jordisala1991** [comment]
For the Array to class change for instance. 

IMO you should list what you did in this PR in the body of the initial message first. This Pr is not only about bumping psalm and phpstan

**Comment by jordisala1991** [comment]
This Pr also overrides another one from @franmomu , should be nice to link them too.

**Comment by VincentLanglet** [comment]
> ```gfm
> ### Changed
> - Use `FilterData` instead of `array` in filters.
> ```

Is this enough ?

**Comment by VincentLanglet** [comment]
> IMO you should list what you did in this PR in the body of the initial message first. This Pr is not only about bumping psalm and phpstan

The PR was initially bumping psalm/phpstan but since it was opened too much time, I had to
- Add the FilterData support in order to be compatible with the latest SonataAdmin commits, I added the changelog.
- Merge 3.x into this branch in order to take some bugfix, this doesn't require a changelog.

**Pull request merged**